static/js/image_upload_widget.js

```diff
@@ -0,0 +1,72 @@
+exports.set_up = function () {
+    function image_upload_complete(spinner, upload_text, delete_button) {
+        spinner.css({visibility: 'hidden'});
+        upload_text.show();
+        delete_button.show();
+    }
+
+    function image_upload_start(spinner, upload_text, delete_button) {
+        spinner.css({visibility: "visible"});
+        upload_text.hide();
+        delete_button.hide();
+    }
+
+    function upload_image(file_input, widget) {
+        const form_data = new FormData();
+        let url;
+        let night_logo = false;
+        if (widget === 'user-avatar-upload-widget') {
+            url = '/json/users/me/avatar';
+        } else if (widget === 'realm-icon-upload-widget') {
+            url = '/json/realm/icon';
+        } else {
+            url = '/json/realm/logo';
+            if (widget === 'realm-night-logo-upload-widget') {
+                night_logo = true;
+            }
+            form_data.append('night', JSON.stringify(night_logo));
+        }
+        form_data.append('csrfmiddlewaretoken', csrf_token);
+        for (const [i, file] of Array.prototype.entries.call(file_input[0].files)) {
+            form_data.append('file-' + i, file);
+        }
+
+        const spinner = $(`#${widget} .upload-spinner-background`).expectOne();
+        const upload_text =  $(`#${widget}  .settings-page-upload-text`).expectOne();
+        const delete_button = $(`#${widget}  .settings-page-delete-button`).expectOne();
+        const error_field = $(`#${widget}  .image_file_input_error`).expectOne();
+        image_upload_start(spinner, upload_text, delete_button);
+        error_field.hide();
+        channel.post({
+            url: url,
+            data: form_data,
+            cache: false,
+            processData: false,
+            contentType: false,
```

Comment:
Not sure of context but seeing your test and this code lines, I suggest to do in your channel.post mock code:
```javascript
assert.equal(req.cache, false);
assert.equal(req.processData, false);
assert.equal(req.contentType, false);
```
This will protect your feature if somebody in future say pass, cache: true but your feature always want cache: false.

Reply:
ok sure